zkvm/src/signature/multikey.rs

```diff
@@ -0,0 +1,58 @@
+use crate::errors::VMError;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+#[derive(Clone)]
+pub struct Multikey {
+    transcript: Transcript,
+    X_agg: VerificationKey,
+}
+
+impl Multikey {
+    pub fn new(pubkeys: Vec<VerificationKey>) -> Result<Self, VMError> {
+        // Create transcript for Multikey
+        let mut transcript = Transcript::new(b"ZkVM.aggregated-key");
+
+        // Hash in pubkeys
+        // L = H(X_1 || X_2 || ... || X_n)
+        for X_i in &pubkeys {
+            transcript.commit_point(b"X_i.L", &X_i.0);
+        }
+
+        let mut multikey = Multikey {
```

Comment:
would it be simpler to create the Multikey closer to the end, so you don't need to have dummy `X_agg`.

Reply:
See discussion here:
https://github.com/interstellar/slingshot/pull/190#discussion_r264976476
I am trying not to deduplicate the `factor_for_key` code, which is why I am creating this half-formed multikey (so I can call the `factor_for_key` function on it

Reply:
`factor_for_key` can internally use factored-out private type-level function `Multikey::compute_factor(&mut Transcript, &VerificationKey)` which we can reuse here w/o a dummy instance.

Reply:
oh whoops, I didn't think of that. Thanks for the suggestion!

(Also, I think we can pass in `&Transcript` instead of `&mut Transcript` to `compute_factor`, since it is only cloning the transcript & modifying the clone, not modifying the original.